### PR TITLE
Add timeout and Don't call cronitor if we pass empty or null monitor code 

### PR DESCRIPTION
**Pull request by ovalente**
Add 10 second time to cronitor call.
Don't try to call cronitor if we don't have a monitor code and log warn when it happend

### CODE REVIEW

File: src/main/java/ca/sm360/cronitor/client/CronitorClient.java [hunk 1]
```diff
@@ -26,44 +30,80 @@ public CronitorClient(String apiKey) {
...
 
     public void run(String monitorCode) throws IOException, URISyntaxException {
-
-        URL commandUrl = new RunCommandUrlGenerator().buildURI(monitorCode, apiKey);
-        cronitorPinger.ping(commandUrl);
+        if(StringUtils.isNotBlank(monitorCode)) {
+            URL commandUrl = new RunCommandUrlGenerator().buildURI(monitorCode, apiKey);
+            cronitorPinger.ping(commandUrl);
+        } else {
+            logger.warning("we can't ping cronitor because the monitor code pass was null or empty");
```
Comment (solutionsmedias360) [inline src/main/java/ca/sm360/cronitor/client/CronitorClient.java]: We can't ping /run on cronitor

File: src/main/java/ca/sm360/cronitor/client/CronitorPinger.java [hunk 2]
```diff
@@ -10,13 +11,22 @@
...
-        connection.getInputStream();
-        connection.disconnect();
+        connection.setConnectTimeout(cronitorPingTimeoutInSecond * 1000);
+
+        try {
+            connection.connect();
+            connection.getInputStream();
+            connection.disconnect();
+        } catch (SocketTimeoutException e) {
+            logger.warning(String.format("fail to call url [%s] because of a timeout trigger after %d seconds",
```
Comment (solutionsmedias360) [inline src/main/java/ca/sm360/cronitor/client/CronitorPinger.java]: Failed to call url [%s] : a timeout occurred after %d seconds

File: src/main/java/ca/sm360/cronitor/client/CronitorPinger.java [hunk 3]
```diff
@@ -10,13 +11,22 @@
...
+        connection.setConnectTimeout(cronitorPingTimeoutInSecond * 1000);
+
+        try {
+            connection.connect();
+            connection.getInputStream();
+            connection.disconnect();
+        } catch (SocketTimeoutException e) {
+            logger.warning(String.format("Failed to call url [%s] : a timeout occurred after %d seconds",
+                    commandURL, cronitorPingTimeoutInSecond));
+        }
```
Comment (firone) [inline src/main/java/ca/sm360/cronitor/client/CronitorPinger.java]: Try catch ignore ??

File: src/main/java/ca/sm360/cronitor/client/CronitorClient.java [hunk 4]
```diff
@@ -26,44 +30,80 @@ public CronitorClient(String apiKey) {
...
 
     public void run(String monitorCode) throws IOException, URISyntaxException {
-
-        URL commandUrl = new RunCommandUrlGenerator().buildURI(monitorCode, apiKey);
-        cronitorPinger.ping(commandUrl);
+        if(StringUtils.isNotBlank(monitorCode)) {
+            URL commandUrl = new RunCommandUrlGenerator().buildURI(monitorCode, apiKey);
+            cronitorPinger.ping(commandUrl);
+        } else {
+            logger.warning("We can't ping /run on cronitor because the monitor code pass was null or empty");
```
Comment (solutionsmedias360) [inline src/main/java/ca/sm360/cronitor/client/CronitorClient.java]: the given monitor code was null or empty

File: src/main/java/ca/sm360/cronitor/client/CronitorClient.java [hunk 5]
```diff
@@ -26,44 +30,80 @@ public CronitorClient(String apiKey) {
...
 
     public void unpause(String monitorCode) throws URISyntaxException, IOException {
-        URL commandUrl = new PauseCommandUrlGenerator(0).buildURI(monitorCode, apiKey);
-        cronitorPinger.ping(commandUrl);
+
+        if(StringUtils.isNotBlank(monitorCode)) {
+            URL commandUrl = new PauseCommandUrlGenerator(0).buildURI(monitorCode, apiKey);
+            cronitorPinger.ping(commandUrl);
+        } else {
+            logger.warning("we can't pause monitor because the monitor code pass was null or empty");
```
Comment (solutionsmedias360) [inline src/main/java/ca/sm360/cronitor/client/CronitorClient.java]: we can't unpause

**Pull request merged**